operations/mimir/servicemonitor.libsonnet

```diff
@@ -0,0 +1,51 @@
+{
+  local this = self,
+
+  serviceMonitor: if $._config.create_service_monitor then {
+    apiVersion: 'monitoring.coreos.com/v1',
+    kind: 'ServiceMonitor',
+    metadata: {
+      name: 'mimir',
+      namespace: $._config.namespace,
+    },
+    spec: {
+      namespaceSelector: {
+        matchNames: [$._config.namespace],
+      },
+      servicesToRelabel:: [
+        'compactor',
+        'distributor',
+        'ingester',
+        'querier',
+        'query-front',
+        'store-gateway',
+      ],
+      endpoints: [
+        {
+          port: port.name,
+          path: '/metrics',
+          [if std.member(self.servicesToRelabel, this[name].spec.selector.name) then 'relabelings']: [{
+            sourceLabels: ['job'],
+            action: 'replace',
+            replacement: $._config.namespace + '/$1',
+            targetLabel: 'job',
+          }],
+        }
+        for name in std.objectFields(this)
+        if std.isObject($[name]) && std.objectHas(this[name], 'kind') && this[name].kind == 'Service'
+        for port in std.filter(function(x) std.length(std.findSubstr('-metrics', x.name)) > 0, this[name].spec.ports)
+      ],
+      selector: {
+        matchExpressions: [{
+          key: 'name',
+          operator: 'In',
+          values: [
+            this[name].spec.selector.name
+            for name in std.objectFields(this)
+            if std.isObject(this[name]) && std.objectHas(this[name], 'kind') && this[name].kind == 'Service' && std.objectHas(this[name].spec.selector, 'name')
```

Comment:
i think this will end up scraping the headless and the non-headless services. For example `query-scheduler` and `query-scheduler-headless` both have the `name` selector.

Reply:
yes, it adds the entry twice, 
```
  selector:
    matchExpressions:
    - key: name
      operator: In
      values:
      - querier
      - query-frontend
      - query-scheduler
      - query-scheduler
```
but it won't select the pods twice, like doing `kubectl get pod -l name=query-scheduler,name=query-scheduler` won't give double entries

Reply:
I added the same label used in the chart to exclude the headless svc: `prometheus.io/service-monitor: "false"`